### PR TITLE
toga-iOS: Fixed imageview to work with iOS. Leave frame to pack for now,

### PR DESCRIPTION
**Pull request by maranas**
Signed-off-by: Moises Aranas <moises.aranas@gmail.com>

This updates the iOS image view to behave more like the Cocoa version.
I made the default scaling behavior to aspect-fit, as that is typical for most
phone implementations, but I need some feedback on this.
I'm leaving alignment and size hinting to the container, so it uses the available
space unless the style's height/width values are set

This also updates Image so the image downloading works.

## PR Checklist:
- [x] All new features have been tested
- [x] All new features have been documented
- [x] I have read the **CONTRIBUTING.md** file
- [x] I will abide by the code of conduct

### CODE REVIEW

**Comment by meichthys** [comment]
related: #78

**Comment by maranas** [comment]
Thanks for checking! BTW should we create a separate toga-demo for iOS?

**Comment by freakboy3742** [comment]
@maranas Ideally, we'd only have a single toga-demo. Toga-demo as currently defined obviously won't work with iOS, so the better approach would be to restructuring toga-demo so that it is a good example of what can/should be done with Toga's "capture the underlying motivation of the UX" philosophy.

Ideally, that would then be an app that we could push to the App Store.

**Comment by maranas** [comment]
@freakboy3742 got it. For this pull request i'm making a specific example for the widget, I don't think i should modify the toga-demo yet since most other targets don't support the imageview widget yet.

File: src/iOS/toga_iOS/widgets/image.py [hunk 1]
```diff
@@ -1,12 +1,24 @@
...
         self.interface = interface
 
     def load_image(self, path):
+        # If a remote URL is provided, use the download from NSData (similar to toga-cocoa)
         if path.startswith('http://') or path.startswith('https://'):
             self.native = UIImage.imageWithData_(NSData.dataWithContentsOfURL_(NSURL.URLWithString_(path)))
         else:
-            self.native = UIImage.alloc().initWithContentsOfFile_(path)
+            basename = os.path.basename(path)
+            if basename == path:
```
Comment (freakboy3742) [inline src/iOS/toga_iOS/widgets/image.py]: Can this differentiate between a named image, and an image that is in the root of the resources directory?

**Comment by maranas** [comment]
UIImages are usually initialized from a name in iOS because when images are built with the app, the images are copied to the root of the bundle - otherwise, you have to provide a full path to it. So it is either a name or a full path. But you have a good point, I shouldn't expose the app_dir in this case. I added this because our current briefcase template bundles everything into the app bundle, respecting the directory tree - so to get the full path to the image, I need to get the full path to the image by appending it to the app_dir path.

I have an idea though - since we will be doing this with iOS classes for other possible widgets (e.g. if we plan to load icons or fonts), we can just add a helper method to resolve relative paths (e.g. the 'resource' directory) to their full paths, hidden from the "public interface" but accessible to the widgets. What do you think?

**Comment by maranas** [comment]
@freakboy3742 I updated it based on the latest comments - now the paths should be treated the same way we do other platforms (so examples/imageview should also work e.g. in gtk). The relative path resolution is now done in the private interface, similar to how we do it in the Icon widget

File: src/iOS/toga_iOS/widgets/image.py (hunk 1 above)
Comment (maranas) [reply src/iOS/toga_iOS/widgets/image.py]: I've removed imageNamed initialization for now, as it goes against normal usage of toga widgets - I was thinking only of iOS when I first implemented it.

**Pull request merged**